pre_commit_hooks/check_branch_name.py

```
@@ -0,0 +1,51 @@
+# branches, and the throwaway names the merge queue creates.
+SKIP_EXACT = {"HEAD", "master", "main", "production"}
+SKIP_PREFIXES = ("gh-readonly-queue/",)
+
+
+def main(argv: Sequence[str] | None = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("filenames", nargs="*")  # accepted & ignored
+    parser.parse_args(argv)
+
+    branch = subprocess.check_output(
+        ["git", "rev-parse", "--abbrev-ref", "HEAD"], text=True
```

Ruff (0.15.17)

[error] Starting a process with a partial executable path

(S607)